docs/use-cases/builds-releases.md

```diff
@@ -0,0 +1,46 @@
+---
+title: "Build & Release"
+---
+
+Convox follows some accepted best practices when it comes to a reliable, dependable and repeatable build process.
+
+- When you [deploy or build](/deployment/builds) your application a "build" artifact is created. This consists of the Docker images that make up your application plus metadata stored in your Rack's database. Each build has a unique ID and is associated with one or more [releases](/docs/releases).
+
+- Every time you deploy an application or change its environment a new [Release](/docs/releases) is created. You can promote any Release for an application to deploy it. Promoting a Release will cause a [rolling update](/docs/rolling-updates) that gracefully deploys the desired Release.
+
+- When a Release is promoted, new processes are gracefully [rolled out](/docs/rolling-updates) into production.
+If there are errors starting new processes, new processes are not [verified as healthy](/application/health-checks), or the rollout doesn't complete in 10 minutes, an automatic [rollback](/deployment/rolling-back) is performed.
+
+- Convox keeps history for your Apps and for Rack and CLI releases. It's possible to [roll back](/deployment/rolling-back) to previous versions of all of these.
+
+All of these steps can be automated and run from CI/CD processes through the Convox CLI!
+
+### Builds
+
+Builds can either be created:
+
+- Manually from the CLI.  `convox build` will create your build artifact, whereas `convox deploy` will create the artifact and also then promote the subsequent release.
+- As part of a Convox [workflow](/console/workflows).  Workflows can connect to your code repository and initiate builds and subsequent activities from code pushes or pull requests.
+- As part of an external CI process.  Convox has a [CircleCI Orb](/external-services/circleci) for easy integration.  Other CI services can also easily integrate with Convox by [installing the Convox CLI](/introduction/installation) in your CI environment, utilising [deploy keys](/console/deploy-keys) to keep your credentials secure, and running the `build` or `deploy` commands as appropriate in your CI workflow.
+
+You can see a list of all the builds for your app by running `convox builds` from the CLI.
+
+### Releases
+
+Releases will automatically be created once a build artifact is created.  They are also created when a change happens to your environment variables, either through the Console or from the CLI.
+
+You can see the list of all releases, and which one is currently active for your app in the Console under the 'Releases' tab, or via the CLI by running `convox releases`.
+
+If running a `convox deploy` then the Release will automatically be promoted with a rolling deployment across your Rack.  Environment variable changes performed from the Console will also be automatically released once you click on 'Apply Changes'.
```

Comment:
"be automatically released" is a bit ambiguous. Release creation and promotion is an important thing to differentiate in this goal doc.

Reply:
Might even want to have a separate section about Promotion, similar to Rollbacks.